### ISSUE TITLE
HamburgerMenu Localization issue

### ISSUE THREAD

**Issue by AndreasReitberger** [MahApps/MahApps.Metro#3369]
**Describe the bug**
I'm using the Hamburger Menu and I wanted to localise the labels next to the icons.
This is one menu item for instance:

```
<!-- HOME -->
                    <Controls:HamburgerMenuGlyphItem Glyph="&#xE8EF;"
                                                     Label="{Globalizer:GlobalizedResource MenuHome, FallbackValue=Home}"
                                                     Tag="main">
                    </Controls:HamburgerMenuGlyphItem>
```

I'm using this extension to handle the localisation.
https://github.com/rhyous/WPFSharp.Globalizer
While the rest of the MainWindow is updating fine on language change, the MenuItems stay untranslated.
Only after a restart (I save the current language and set it in the App.xml file) the MenuItems are translated.
Is there any way to invalidate the HamberMenu and force it to redraw the menuitems?

Thanks a lot!

**Comment by punker76** [comment]
@AndreasReitberger Can you maybe create a short sample together with this library? Thx

**Comment by prkrmx** [comment]
hi @AndreasReitberger 
Did you find a solution? I have the same issue 
Thanks

**Comment by AndreasReitberger** [comment]
Hi @prkrmx 
Sadly only with a bypass in the code behind.
I did this to translate the Hamburger Menu.

`foreach(HamburgerMenuGlyphItem item in MenuControl.Items)
            {
                UserControl uc = new UserControl();
                uc.Tag = item.Tag;
                string tag = item.Tag as string;
                if (tag == null)
                {
                    tag = uc.Tag.ToString();
                    tag = tag.Substring(tag.LastIndexOf(".") + 1);
                }
                switch (tag)
                {
                    case "main":
                        item.Label = Design.getLocalisedStringFromKey("MenuHome");
                        break;
                    case "manage":
                        item.Label = Design.getLocalisedStringFromKey("MenuManage");
                        break;
                    case "jobs":
                        item.Label = Design.getLocalisedStringFromKey("MenuJobs");
                        break;
                    default:
                        throw new Exception("Out of range");
                        break;
                }

                item.InvalidateProperty(HamburgerMenuGlyphItem.LabelProperty);
            }
            //UPDATE OPTIONS ITEMS
            foreach (HamburgerMenuGlyphItem item in MenuControl.OptionsItems)
            {
                UserControl uc = new UserControl();
                uc.Tag = item.Tag;
                string tag = item.Tag as string;
                if (tag == null)
                {
                    tag = uc.Tag.ToString();
                    tag = tag.Substring(tag.LastIndexOf(".") + 1);
                }
                switch (tag)
                {
                    case "licence":
                        item.Label = Design.getLocalisedStringFromKey("MenuLicense");
                        break;
                    case "settings":
                        item.Label = Design.getLocalisedStringFromKey("MenuSettings");
                        break;
                    case "about":
                        item.Label = Design.getLocalisedStringFromKey("MenuAbout");
                        break;
                }

                item.InvalidateProperty(HamburgerMenuGlyphItem.LabelProperty);
            }
        }`

Not nice, but working. Hope it helps.
Best,
Andreas

**Comment by timunie** [comment]
Hi @AndreasReitberger , @punker76 , @prkrmx ,

I had a look at the provided sample just for fun. The good news is, that I maybe found a Solution for the Problem. 

1.) This did **not** the trick: I updated to the latest Alpha version to check if the problem is already gone.
2.) I changed the `HamburgerMenuItem` so that `Label` derives from `object` instead of `string`. The `Label` itself is then: 
```xaml 
<test:HamburgerMenuGlyphItem Glyph="&#xE8EF;" Tag="main">
    <test:HamburgerMenuGlyphItem.Label>
        <TextBlock Text="{Globalizer:GlobalizedResource MenuLicense, FallbackValue=License}" />
    </test:HamburgerMenuGlyphItem.Label>
</test:HamburgerMenuGlyphItem>
``` 
With this change I can change the Language dynamically. 

I am not sure if I am really on the right track, but my personal conclusion is: `HamburgerMenuItem` doesn't derive from `FrameworkElement`. This may cause this behavior. Maybe we should change the base class from `Freezeable` to `ListBoxItem`, but this is just a guess. 

Happy coding
Tim
